dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -107,6 +106,7 @@ class Loader {
     // perform an iniital buffer fill if it hasn't already happened
     if (!initial_buffer_filled_) {
       TimeRange tr("[Loader] Filling initial buffer", TimeRange::kBlue1);
+      std::lock_guard<std::mutex> lock(return_mutex_);
```

Comment:
Maybe we should add a comment that this is about `empty_tensors_` couple lines later? So we remember later

Reply:
what comment do you have in mind?